### PR TITLE
add did jwt vc

### PR DESCRIPTION
**Pull request by aldigjo**

### CODE REVIEW

File: src/Credentials.ts [hunk 1]
```diff
@@ -163,13 +169,16 @@ interface VerificationParam {
   exp?: number
   vc?: string[]
   callbackUrl?: string
+  nbf?: number,
+  expiresIn?: number
```
Comment (mirceanis) [inline src/Credentials.ts]: is this meant to align the API for different requests?

Comment (aldigjo) [reply src/Credentials.ts]: correct, since the new did-jwt requires expiresIn

File: src/Credentials.ts [hunk 2]
```diff
@@ -838,10 +877,35 @@ class Credentials {
...
+  createPresentationRequest(params: PresentationRequestParams, expiresIn = 600) {
+    const payload: PresentationRequest = {}
+    if (params.iss) payload.iss = params.iss
+    if (params.claims) payload.claims = params.claims
+    if (params.callbackUrl) payload.callback = params.callbackUrl
+
+
+    return this.signJWT(
+      { ...payload, type: Types.PRESENTATION_REQUEST },
+      expiresIn ? undefined : expiresIn
```
Comment (mirceanis) [inline src/Credentials.ts]: If I'm reading this right, then expiresIn never gets passed in unless it's "falsey" which is probably not the intent

File: src/Credentials.ts [hunk 3]
```diff
@@ -523,7 +560,7 @@ class Credentials {
    * @param    {Number}      [opts.expiresIn]    The duration in seconds after which the request expires
    * @returns  {Promise<Object, Error>}          A promise which resolves with a signed JSON Web Token or rejects with an error
    */
-  createVerificationSignatureRequest(unsignedClaim: object, { aud, sub, riss, callbackUrl, vc, rexp, expiresIn}:VerificationRequest) {
+  createVerificationSignatureRequest(unsignedClaim: object, { aud, sub, riss, callbackUrl, vc, rexp, expiresIn, nbf }:VerificationRequest) {
```
Comment (mirceanis) [inline src/Credentials.ts]: is `nbf` required?

File: src/__tests__/Credentials-test.ts [hunk 4]
```diff
@@ -679,9 +697,125 @@ describe('txRequest()', () => {
...
+        type: ['VerifiableCredential'],
+        verifiableCredential: [vcJwt]
+      }
+    }
+
+    const issuer: Issuer = { did: credentials.did, signer: credentials.signer }
+    const vpJwt = await createPresentation(vpPayload, issuer)
+
+    const result = await credentials.verifyPresentation(vpJwt)
+    expect(result.payload.vp).toEqual(vpPayload.vp)
```
Comment (mirceanis) [inline src/__tests__/Credentials-test.ts]: this or another test should also check for the other fields in the payload, not just `.vp`

File: src/Credentials.ts [hunk 5]
```diff
@@ -838,10 +875,33 @@ class Credentials {
    *  @return   {Promise<Object, Error>}                        a promise which resolves with a parsed response or rejects with an error.
    */
   async verifyDisclosure(token: string) {
-    const { payload, doc } = await verifyJWT(token, { audience: this.did })
+    const { payload, doc } = await verifyJWT(token, { resolver: this.resolver, audience: this.did })
     return this.processDisclosurePayload({ payload, doc })
   }
 
+  createPresentationRequest(params: PresentationRequestParams, expiresIn = 600) {
```
Comment (mirceanis) [inline src/Credentials.ts]: `PresentationRequestParams` is supposed to be a simplified version of `DisclosureRequestPayload`, right?

@simonas-notcat does this comply with the current expectations of daf / daf-mobile?
Does daf-mobile answer `shareReq` with Presentations already or is it expecting a specific `type` in the payload?
Please weigh in here, I want to make sure we're not forking the specs here.

**Pull request merged**